scheduler/src/main/scala/com.gu.media.scheduler/SchedulerLambda.scala

```diff
@@ -0,0 +1,70 @@
+package com.gu.media.scheduler
+
+import java.time.Instant
+import java.time.temporal.ChronoUnit
+
+import com.amazonaws.services.lambda.runtime.{Context, RequestHandler}
+import com.gu.contentatom.thrift.atom.media.PrivacyStatus
+import com.gu.media.CapiAccess
+import com.gu.media.lambda.LambdaBase
+import com.gu.media.logging.Logging
+import com.gu.media.youtube.{YouTubeAccess, YouTubeVideos}
+import play.api.libs.json.{JsArray, JsValue}
+
+import scala.annotation.tailrec
+import scala.util.control.NonFatal
+
+class SchedulerLambda extends RequestHandler[Unit, Unit]
+  with LambdaBase
+  with Logging
+  with CapiAccess
+  with YouTubeAccess
+  with YouTubeVideos {
+
+  override def handleRequest(input: Unit, context: Context): Unit = {
+    val now = Instant.now()
+    val oneDayAgo = Instant.now().minus(1, ChronoUnit.DAYS)
+    val assets = getVideosFromScheduledAtoms(1, 100, oneDayAgo, now, Set.empty).filter(isManagedVideo)
+
+    assets.foreach { video =>
+      try {
+        setStatus(video, PrivacyStatus.Public)
+      } catch {
+        case NonFatal(err) =>
+          log.error(s"Unable to launch $video", err)
+      }
+    }
+  }
+
+  @tailrec
+  private def getVideosFromScheduledAtoms(page: Int, pageSize: Int, oneDayAgo: Instant, now: Instant, before: Set[String]): Set[String] = {
+    val url = s"atoms?types=media&page-size=$pageSize&page=$page&from-date=$oneDayAgo&to-date=$now&use-date=scheduled-launch"
+    val response = (capiQuery(url) \ "response").get
+    val currentPage = (response \ "currentPage").as[Int]
+    val pages = (response \ "pages").as[Int]
+
+    val results = (response \ "results").as[JsArray].value
+    val after = results.foldLeft(before) { (acc, atom) => acc ++ getScheduledVideos(atom) }
+
+    if(currentPage < pages)
+      getVideosFromScheduledAtoms(page + 1, pageSize, oneDayAgo, now, after)
+    else
+      after
+  }
+
+  private def getScheduledVideos(atom: JsValue): Set[String] = {
+    val scheduledLaunchDate = (atom \ "contentChangeDetails" \ "scheduledLaunch" \ "date").as[Long]
```

Comment:
using the model should make things a bit easier:

```scala
import com.gu.contentatom.thrift.atom.media.{MediaAtom => ThriftMediaAtom}
import com.gu.media.model MediaAtom

val thriftAtom: ThriftMediaAtom = atom.validate[ThriftMediaAtom]
val mediaAtom: MediaAtom = MediaAtom.fromThrift(thriftAtom)
val scheduledLaunchDate: Option[ChangeRecord] = mediaAtom.contentChangeDetails.scheduledLaunch
...
```

However, I realise we're using `JsValue` elsewhere, so feel free to do this in a separate PR.